### ISSUE TITLE
Option to replace site name text + logo by a picture

### ISSUE THREAD

**Issue by tdavidcl** [squidfunk/mkdocs-material#6417]
### Context

Currently the layout in the header is : 
| logo | title | ....

### Description

It would be nice to have also an option in the config to do : 
| logo + title as a picture | ...
, where the logo+title picture is clickable as the text logo currently.

Some projects tends to have a logo that also include the name of the project (for ex : [AdaptiveCpp](https://github.com/AdaptiveCpp/AdaptiveCpp), [Shamrock](https://github.com/Shamrock-code/Shamrock-units)). Which make the current layout of the header unsuitable.

Also since it may increase the taller than just title as text, it would be nice to be able to increase the height of the header in the config.

### Related links

 - [Header configuration in doc](https://squidfunk.github.io/mkdocs-material/setup/setting-up-the-header/)
 - [AdaptiveCpp](https://github.com/AdaptiveCpp/AdaptiveCpp)

### Use Cases

It would make adoption easier for projects with such logo (that include the name) , instead of relying on tweaks in the HTML theme.

### Visuals

 - [AdaptiveCpp logo](https://github.com/AdaptiveCpp/AdaptiveCpp/blob/develop/doc/img/logo/logo-color.png?raw=true)
 - [Shamrock logo](https://github.com/Shamrock-code/Shamrock-units/blob/main/doc/shamrock-doc/src/images/no_background_nocolor.png?raw=true)

### Before submitting

- [X] I have read and followed the [change request guidelines](https://squidfunk.github.io/mkdocs-material/contributing/requesting-a-change/).
- [X] I have verified that [my idea is a change request and not a bug report](https://squidfunk.github.io/mkdocs-material/contributing/requesting-a-change/#its-not-a-bug-its-a-feature).
- [X] I have ensured that, to the best of my knowledge, [my idea will benefit the entire community](https://squidfunk.github.io/mkdocs-material/contributing/requesting-a-change/#benefit-for-the-community).
- [X] I have included relevant links to [the documentation](https://squidfunk.github.io/mkdocs-material/), related [issues](https://github.com/squidfunk/mkdocs-material/issues), and [discussions](https://github.com/squidfunk/mkdocs-material/discussions) to underline the need for my idea.

**Comment by squidfunk** [comment]
> Also since it may increase the taller than just title as text, it would be nice to be able to increase the height of the header in the config.

On a side note: the height increases automatically. The whole CSS and JavaScript is designed to account for customizations, so I'm pretty sure that your goals should be easily achievable.

**Comment by tdavidcl** [comment]
Yup, I managed to make it work nicely. As you said the css is sufficiently flexible for that, it was simple.
I was just thinking that this may be sufficiently widespread (tall logo or hybrid text+logo) to be in the config rather than through tweaks only, but project size is clearly a valid concern 😅.

**Comment by itskalvik** [comment]
@tdavidcl Can you please explain how you got this working? I am trying to setup a similar logo/title image and I am not sure how to set it up 😅
